core/src/main/scala/laserdisc/protocol/KeyP.scala

```diff
@@ -94,20 +96,21 @@ trait KeyBaseP {
 
   import keytypes._
 
-  private[this] implicit final val str2NOKEYOrOK: Str ==> (NOKEY | OK) = Read.instancePF {
-    case Str("NOKEY") => Left(NOKEY)
-    case Str("OK")    => Right(OK)
+  private[this] implicit final val str2NOKEYOrOK: Str ==> (NOKEY | OK) = Read.instance {
+    case Str("NOKEY") => Right(Left(NOKEY))
+    case Str("OK")    => Right(Right(OK))
+    case Str(other)   => Left(Err(s"Unexpected string for Str ==> (NOKEY | OK). Was $other"))
   }
 
   private[this] final val zeroIsNone = RESPRead.instance(Read.numZeroIsNone[PosInt])
 
-  private[this] implicit final val str2OptionType: Str ==> Option[KeyType] = Read.instancePF {
-    case Str("none")   => None
-    case Str("string") => Some(KeyType.string)
-    case Str("list")   => Some(KeyType.list)
-    case Str("set")    => Some(KeyType.set)
-    case Str("zset")   => Some(KeyType.zset)
-    case Str("hash")   => Some(KeyType.hash)
+  private[this] implicit final val str2OptionType: Str ==> Option[KeyType] = Read.instance {
+    case Str("string") => Right(Some(KeyType.string))
+    case Str("list")   => Right(Some(KeyType.list))
+    case Str("set")    => Right(Some(KeyType.set))
+    case Str("zset")   => Right(Some(KeyType.zset))
+    case Str("hash")   => Right(Some(KeyType.hash))
+    case Str(_)        => Right(None)
```

Comment:
isn't this changed semantics from before?
```suggestion
    case Str("none")   => Right(None)
    case Str(other)    => Left(Err(s"Unexpected string for Str ==> Option[KeyType]. Was $other"))
```